packages/image_picker/image_picker/ios/Classes/FLTPHPickerSaveItemToPathOperation.m

```diff
@@ -0,0 +1,206 @@
+  return executing;
+}
+
+- (BOOL)isFinished {
+  return finished;
+}
+
+- (void)setFinished:(BOOL)isFinished {
+  [self willChangeValueForKey:@"isFinished"];
+  self->finished = isFinished;
+  [self didChangeValueForKey:@"isFinished"];
+}
+
+- (void)setExecuting:(BOOL)isExecuting {
+  [self willChangeValueForKey:@"isExecuting"];
+  self->executing = isExecuting;
+  [self didChangeValueForKey:@"isExecuting"];
+}
+
+- (void)completeOperationWithPath:(NSString *)savedPath {
+  [self setExecuting:NO];
+  [self setFinished:YES];
+  getSavedPath(savedPath);
+}
+
+- (void)processImage API_AVAILABLE(ios(14)) {
+  [self.result.itemProvider
+      loadObjectOfClass:[UIImage class]
+      completionHandler:^(__kindof id<NSItemProviderReading> _Nullable data,
+                          NSError *_Nullable error) {
+        if ([data isKindOfClass:[UIImage class]]) {
+          __block UIImage *localImage = data;
+          PHAsset *originalAsset = [self getAssetFromPHPickerResult:self.result];
+          if (self.maxWidth != (id)[NSNull null] || self.maxHeight != (id)[NSNull null]) {
+            localImage = [FLTImagePickerImageUtil scaledImage:localImage
+                                                     maxWidth:self.maxWidth
+                                                    maxHeight:self.maxHeight
+                                          isMetadataAvailable:originalAsset != nil];
+          }
+          __block NSString *savedPath;
+          if (!originalAsset) {
+            // Image picked without an original asset (e.g. User picked an image without
+            // permission).
+            savedPath = [self saveImageWithPickerInfo:nil
+                                                image:localImage
+                                         imageQuality:self.desiredImageQuality];
+            [self completeOperationWithPath:savedPath];
+          } else {
+            [[PHImageManager defaultManager]
+                requestImageDataForAsset:originalAsset
+                                 options:nil
+                           resultHandler:^(NSData *_Nullable imageData, NSString *_Nullable dataUTI,
+                                           UIImageOrientation orientation,
+                                           NSDictionary *_Nullable info) {
+                             // maxWidth and maxHeight are used only for GIF images.
+                             savedPath =
+                                 [self saveImageWithOriginalImageData:imageData
+                                                                image:localImage
+                                                             maxWidth:self.maxWidth
+                                                            maxHeight:self.maxHeight
+                                                         imageQuality:self.desiredImageQuality];
+                             [self completeOperationWithPath:savedPath];
+                           }];
+          }
+        }
+      }];
+}
```

Comment:
This method was already in here, it's simply moved over from `FLTPHPickerSaveImageToPathOperation`.